app/controllers/samples_controller.rb

```diff
@@ -802,13 +802,25 @@ def report_v2
     pipeline_run = select_pipeline_run(@sample, params[:pipeline_version])
     background_id = get_background_id(@sample, params[:background])
 
-    skip_cache = params[:skip_cache] || false
+    # Don't cache the response until the pipeline run is report-ready
+    # so the displayed pipeline run status will be updated correctly.
+    skip_cache = !pipeline_run.report_ready? || params[:skip_cache] || false
```

Comment:
Do you think we still need this with the report ts fix?

Reply:
I guess we can keep it.

Reply:
The `report_ts` fix should prevent the report from always showing the same loading screen, but I don't think it makes much sense to cache the report unless it's ready anyways.

Reply:
Let's keep it.